### PR TITLE
feat(ticket): remove pending status from follow-up/task

### PR DESCRIPTION
**Pull request by Rom1-B**
Before GLPI 10.0, we could define the status of the ticket when adding a follow-up or a task.

image

Since, GLPI 10.0 it is more automatic, and we can only add the status "pending" not remove it, for that it is necessary to do it by updating the ticket header.

This PR activates the toggle if the ticket is already pending, and if you deactivate it, the ticket becomes assigned or new (if there is no technician)

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !24936

### CODE REVIEW

File: src/CommonITILTask.php [hunk 1]
```diff
@@ -635,6 +637,45 @@ public function post_addItem()
 
            // Set parent status to pending
             $this->input['_status'] = CommonITILObject::WAITING;
+        } elseif ($this->input['_job']->fields["status"] == CommonITILObject::WAITING) {
```
Comment (orthagh) [inline src/CommonITILTask.php:640]: I think all this part should be factorized as it's almost the same as Followup

**Review by trasher — APPROVED** [review]

File: src/CommonDBTM.php [hunk 2]
```diff
@@ -6508,4 +6508,126 @@ public function isGlobal(): bool
 
         return $is_global;
     }
+
```
Comment (orthagh) [inline src/CommonDBTM.php:6511]: good for the factorisation but not in commonDBTM please.
As said in GLPI mattermost channel today, a trait could be the way

**Review by orthagh — CHANGES_REQUESTED** [review]

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

File: src/Features/ParentStatus.php [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+use Session;
+
+/**
+ * ParentStatus
+ *
+ * @since 10.0.4
+ */
+trait ParentStatus
+{
+    public function updateParentStatus(): void
```
Comment (cedric-anne) [inline src/Features/ParentStatus.php:50]: Changes made to `$this->input` seems to be used only in this new method. Passing values in parameters would make static code analysis easier for IDEs and code analysis tools.

```suggestion
    public function updateParentStatus(CommonITILObject $parentitem, array $input): void
```

**Review by trasher — APPROVED** [review]

File: src/Features/ParentStatus.php [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+/**
+ * ParentStatus
+ *
+ * @since 10.0.4
+ */
+trait ParentStatus
+{
+    public function updateParentStatus(CommonITILObject $parentitem, array $input): void
+    {
+        $parentitem = $this->input['_job'];
```
Comment (cedric-anne) [inline src/Features/ParentStatus.php:52]:
```suggestion
```

File: src/Features/ParentStatus.php [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        ) {
+            $update = [
+                'id'        => $parentitem->fields['id'],
+                'status'    => CommonITILObject::CLOSED,
+                'closedate' => $_SESSION["glpi_currenttime"],
+                '_accepted' => true,
+            ];
+
+           // Use update method for history
+            $input["_job"]->update($update);
```
Comment (cedric-anne) [inline src/Features/ParentStatus.php:82]:
```suggestion
            $parentitem->update($update);
```

File: src/Features/ParentStatus.php [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+                $parentitem->update($update);
+                $reopened     = true;
+            }
+        }
+
+        if (
+            !empty($this->fields['begin'])
+            && $input["_job"]->isStatusExists(CommonITILObject::PLANNED)
+            && (($input["_job"]->fields["status"] == CommonITILObject::INCOMING)
+              || ($input["_job"]->fields["status"] == CommonITILObject::ASSIGNED)
```
Comment (cedric-anne) [inline src/Features/ParentStatus.php:149]:
```suggestion
            && $parentitem->isStatusExists(CommonITILObject::PLANNED)
            && (($parentitem->fields["status"] == CommonITILObject::INCOMING)
              || ($parentitem->fields["status"] == CommonITILObject::ASSIGNED)
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**